prow/apis/prowjobs/v1/types.go

```diff
@@ -621,6 +625,21 @@ func (g *GCSConfiguration) Validate() error {
 	return nil
 }
 
+// ParseBucketName tries to extract the storageProvider and bucket name from, e.g.:
+// * <bucket-name> (storageProvider gs)
+// * <storage-provider>://<bucket-name>
+func ParseBucketName(bucket string) (string, string, error) {
+	// default to GCS if no storage-provider is specified
+	if !strings.Contains(bucket, "://") {
+		bucket = "gs://" + bucket
+	}
+	parsedBucket, err := url.Parse(bucket)
+	if err != nil {
+		return "", "", fmt.Errorf("bucket field %q has invalid format, expected either <bucket-name> or <storage-provider>://<bucket-name>", bucket)
+	}
+	return parsedBucket.Scheme, parsedBucket.Host + parsedBucket.Path, nil
```

Comment:
Shouldn't this be `Host + "/" + Path`? Are we unit testing this?

Reply:
Yup that's unit tested in `TestParseBucket`. Looks like Path contains the leading slash, e.g.:
```
gs://my-floppy-backup/a://doom2.wad.006
Scheme: gs
Host: my-floppy-backup
Path: /a://doom2.wad.006
```